File: Kushagra-PB002/wikiSearch ruby/wiki.rb
require 'open-uri'
require "rubygems"
require "json"

$api_url = "https://en.wikipedia.org/w/api.php?action=opensearch&limit=1&format=json&search="

class WikiSearch

	def initialize(query)
		@query = query
	end

	def searchWiki(logfile)
		logfile = open(logfile, "a")
		search_results = open($api_url + @query).read
		results = JSON.parse(search_results)
		logfile.puts " #{Time.new} #{results[0]} #{results[1]} #{results[3]} "
		return results[3]
	end
end

input_array = ARGV

if input_array.length == 1
	logfile = input_array[0]
	choice = 'y'
	while choice == 'y' or choice =='Y' do
		puts "Enter the keyword to be searched:"
		query = STDIN.gets.to_s
		puts "Searching...  #{query}"
		search_query = WikiSearch.new(query)
		search_link = search_query.searchWiki(logfile)
		puts "Search Completed..."
		puts "Link: #{search_link}"
		puts "Want to enter more queries(y/n):"
        choice = STDIN.gets.to_s.chomp
	end
else 
	logfile = input_array[input_array.length-1]
	for i in 0...(input_array.length-1)
		query = input_array[i]
		puts "Searching...  #{query}"
		search_query = WikiSearch.new(query)
		search_link = search_query.searchWiki(logfile)
		puts "Search Completed..."
		puts "Link: #{search_link}"
	end
end

File: Kushagra-PB002/wikiSearch python/wiki.py
import argparse
import sys
import requests
from datetime import datetime

api_url = "https://en.wikipedia.org/w/api.php?action=opensearch&limit=1&format=json&search="

def searchWiki(query, logfile):
	logfile = open(logfile, "a")
	search_results = requests.get(api_url + query)
	results = search_results.json()
	timestmp = datetime.now().strftime('%Y-%m-%d %H:%M:%S')
	logfile.write(timestmp + "  " + results[0] + "  " + results[1][0] + "  " + results[3][0] + "\n")
	return results[3][0]

parser = argparse.ArgumentParser()
parser.add_argument("logfile", help = "Logs results to this file")
parser.add_argument("query", nargs = "*", help = "Query to be searched for no query starts interactive mode")
args = parser.parse_args()

input_array = sys.argv

if len(input_array) == 2:
	logfile = input_array[1]
	choice = "y"
	while choice =="y" or choice == "Y":
		query = raw_input("Enter the keyword to be searched: ")
		print "Searching...." + query
		try:
			search_link = searchWiki(query, logfile)
		except Exception, e:
			print e
			print "Exiting..."
			sys.exit()
		print "Search Completed"
		print "Link: " + search_link + "\n"
		choice = raw_input("Enter the choice(y/n): ")
else:
	logfile = input_array[1]
	for i in range(2, len(input_array)):
		query = input_array[i]
		print "Searching...." + query
		try:
			search_link = searchWiki(query, logfile)
		except Exception, e:
			print e
			print "Exiting..."
			sys.exit()
		print "Search Completed"
		print "Link: " + search_link + "\n"





